src/platform/update/version.h

```diff
@@ -0,0 +1,60 @@
+/*
```

Comment:
Semantic version is quite standard, so we might be better off offloading to some existing implementation. Here are a couple of examples that came up in a search: https://github.com/zmarko/semver and https://github.com/Neargye/semver

Reply:
@Saviq asked more for the dpkg versioning scheme though...

Reply:
OK, I may have missed things, let's discuss in the meeting

Reply:
Used the latter (former didn't interpret the spec for prerelease the way we needed - it only accepts "alpha", "beta" and a few others, not strings like the spec says).
I had to patch the lib to fix some integration issues, patches proposed to upstream

Reply:
I think you switched latter/former in the comment but glad to know this was offloaded.